Makefile

```
@@ -27,13 +33,44 @@ run: start-llama-stack-container ## Run the service locally with dependent servi
+	fi
+	@if ! $(CONTAINER_RUNTIME) image inspect $(LLAMA_STACK_DEPS_IMAGE) >/dev/null 2>&1; then \
+		echo "Deps image not found, building..."; \
+		$(MAKE) build-llama-stack-deps-image; \
+	elif [ "$(CURRENT_DEPS_HASH)" != "$(STORED_DEPS_HASH)" ]; then \
+		echo "Dependencies changed (pyproject.toml or uv.lock), rebuilding deps image..."; \
+		$(MAKE) build-llama-stack-deps-image; \
+	else \
+		echo "✓ Deps image is up-to-date (skipping rebuild)"; \
+	fi
+
+build-llama-stack-image: ensure-llama-stack-deps-image ## Build llama-stack app image (source-only layer on top of deps)
```

checkmake (0.3.2)

[warning] Target body for "build-llama-stack-image" exceeds allowed length of 5 lines (8).

(maxbodylength)